src/main/java/com/limechain/trie/structure/nibble/Nibbles.java

```diff
@@ -92,191 +84,148 @@ public boolean startsWith(Nibbles prefix) {
         return prefix.nibbles.equals(this.nibbles.subList(0, prefixSize));
     }
 
-    @NotNull
-    @Override
-    public Iterator<Nibble> iterator() {
-        return nibbles.iterator();
-    }
-
     /**
      * @return the lower hexadecimal string representation of this Nibbles
      */
     public String toLowerHexString() {
         return toLowerHexString(this.nibbles);
     }
 
-    // TODO:
-    //  This methods exists simply because to Java, not any `List<Nibble>` is the same as `Nibbles`
-    //  Although that's exactly what I want :D
-    private static String toLowerHexString(List<Nibble> nibbles) {
-        return nibbles.stream()
-                .map(Nibble::asLowerHexDigit)
-                .collect(
-                        Collector.of(
-                                StringBuilder::new,
-                                StringBuilder::append,
-                                StringBuilder::append,
-                                StringBuilder::toString
-                        )
-                );
-    }
-
-    @Override
-    public String toString() {
-        return this.toLowerHexString();
-    }
-
-    @Override
-    public Stream<Nibble> stream() {
-        return nibbles.stream();
-    }
-
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        Nibbles nibbles1 = (Nibbles) o;
-        return this.nibbles.equals(nibbles1.nibbles);
-    }
-
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(this.nibbles);
-    }
-
-    @NotNull
-    @Override
-    public Object[] toArray() {
-        return nibbles.toArray();
-    }
-
-    @NotNull
-    @Override
-    public <T> T[] toArray(@NotNull T[] a) {
-        return nibbles.toArray(a);
-    }
-
-    @Override
-    public boolean add(Nibble nibble) {
-        return nibbles.add(nibble);
-    }
-
-    @Override
-    public boolean remove(Object o) {
-        return nibbles.remove(o);
-    }
-
-    @Override
-    public boolean containsAll(@NotNull Collection<?> c) {
-        return new HashSet<>(nibbles).containsAll(c);
-    }
-
-    @Override
-    public boolean addAll(@NotNull Collection<? extends Nibble> c) {
-        return nibbles.addAll(c);
-    }
-
-    @Override
-    public boolean addAll(int index, @NotNull Collection<? extends Nibble> c) {
-        return nibbles.addAll(c);
-    }
-
-    @Override
-    public boolean removeAll(@NotNull Collection<?> c) {
-        return nibbles.removeAll(c);
+    private static String toLowerHexString(Iterable<Nibble> nibbles) {
```

Comment:
I don't like this method being in here, but couldn't figure out a better place. If you think it's ok, let it be.

Reply:
We decided that `BytesToNibbles` and the reverse one could be in a single utility class where it would also make sense to put this.

Reply:
Take a look and tell me whether you like it: a3f60bfc6ac55054ac4f19506131a7652eca6ca9
Resolve if considered finished.